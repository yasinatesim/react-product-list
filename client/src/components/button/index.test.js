/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import Button from './index';

const _onClick = () => {
  // eslint-disable-next-line
  console.log('this is a test');
};

describe('button component', () => {
  it('normal case', () => {
    const tree = renderer.create(<Button onClick={_onClick}>Text</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('no trigger event', () => {
    const tree = renderer.create(<Button>Text</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('no children', () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
