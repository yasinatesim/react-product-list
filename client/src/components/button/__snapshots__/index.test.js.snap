// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button component no children 1`] = `
<button
  onClick={[Function]}
  type="button"
/>
`;

exports[`button component no trigger event 1`] = `
<button
  onClick={[Function]}
  type="button"
>
  Text
</button>
`;

exports[`button component normal case 1`] = `
<button
  onClick={[Function]}
  type="button"
>
  Text
</button>
`;

exports[`renders correctly 1`] = `
<button
  onClick={[Function]}
  type="button"
>
  Text
</button>
`;
